### ISSUE TITLE
Native support for Braille Me display

### ISSUE THREAD

**Issue by xtstoll** [nvaccess/nvda#11497]
### Is your feature request related to a problem? Please describe.
Currently, the Braille Me display does not work with NVDA (2020.2) because the NVDA add-on does not work anymore. It's very problematic because the Braille Me is an accessible, low-cost braille display from India being used in many developing countries by the same public that uses NVDA. Personally, my blind daughter can no longer use her Braille Me here in Brazil with NVDA, which is her primary desktop screen reader.

### Describe the solution you'd like
Native support for the Braille Me display by NVDA.

### Describe alternatives you've considered
Upgrading the Braille Me add-on to work with the latest version.

### Additional context
I believe this issue is related: #8659

Screenshot of error message:
![brailleme_NVDAerror](https://user-images.githubusercontent.com/24814666/90322116-a64c7000-df26-11ea-87f6-61bc978ae892.JPG)

**Comment by alexarnaud** [comment]
Hello @xtstoll ,

Thanks for reporting this to NVDA.

I agree this should be compatible with NVDA.

In the meantime, do you have tried to install BRLTTY which is compatible with the Inceptor Brailleme, see there: http://mielke.cc/brltty/doc/Manual-BRLTTY/English/BRLTTY-8.html

Best regards.

**Comment by leonardder** [comment]
If you could provide any braille display protocol documentation from the manufacturer, it would be possible to look into this. Haven't attempts been made to port the current driver to Python 3?
@feerrenrut Would it be possible to gather info from user stats reporting to see how often this driver is/was used?

**Comment by feerrenrut** [comment]
>Would it be possible to gather info from user stats reporting to see how often this driver is/was used?

Over a 180 day period prior to 2019-08-01, there was a max of 3 BrailleMe users who uploaded user stats with their daily update checks. The average was 1.
Prior to 2018-08-01 there were none.

That said, the number of BRLTTY users has increased.

The addon "BrailleME Display" should ideally be updated by 'Innovision Tech'. Users should contact them to ask if they plan to update it. Otherwise, the addon could be updated by someone else. Without a display to test on it is hard to verify the work. As @leonardder mentioned, with the appropriate information a built in driver could be developed.

**Comment by alexarnaud** [comment]
@leonardder If it works with BRLTTY, maybe it will be possible to check what BRLTTY does to write a driver for NVDA. You could also probably ask the BRLTTY mailing list to get the relevant information.

It's a bit off-topic but I don't understand why NVDA doesn't use BRLTTY instead of developing its own driver. It'd require less maintenance for all, isn't it?

**Comment by leonardder** [comment]
BRLTTY doesn't support many braille displays using their native USB driver, only using LibUSB. I think that's one of the main reasons that holds us back from using it more broadly.
It should be possible to reverse engineer a driver from BLRTTY, but that takes a major amount of time and is impossible to do without a physical display.

**Comment by alexarnaud** [comment]
@leonardder Why is it an issue to not use the native driver? I ask this question because on Linux we use it and it works. Except if I'm wrong, Brailleback on Android or the Narrator use BRLTTY and it works too.

**Comment by lukaszgo1** [comment]
The problem with BRLTTy drivers lies in the fact that they has to be installed by someone and for portable versions of NVDA this cannot be done automatically. Also @alexarnaud I don't know if you've ever tried to install them under  Windows but because they're unsigned this is quite an adventure not something which can be done by less tech savvy users since often Secure Boot has to be disabled in the UEFI to make install work.

**Comment by josephsl** [comment]
Hi, another advantage for using a native driver: automatic braille display detection and assigning NVDA specific hot keys. Thanks.

**Comment by xtstoll** [comment]
@feerrenrut 
I fully agree about having Innovision update their add-on. I sent them an e-mail in March 2020 and again this week. I also reached out the National Braille Press in July, when I didn't get a response from Innovision.

Honestly, at this point, I'm after whoever can help. Here in Brazil, I have an 11-year-old blind daughter study at home during the pandemic, so the braille display is her lifeline for studying and learning language properly.

**Comment by gerald-hartig** [comment]
@michaelDCurran Can this be closed as resolved due to https://github.com/nvaccess/Internal-issue-tracking/issues/246 ?
